includes/namespace.php

```diff
@@ -29,63 +29,111 @@
 use WP_Error;
 use WP_Site;
 
-if (
-	! class_exists( '\Google\Web_Stories\Plugin' ) ||
-	! file_exists( WEBSTORIES_PLUGIN_DIR_PATH . '/assets/js/edit-story.js' )
-) {
-	/**
-	 * Displays an admin notice about why the plugin is unable to load.
-	 *
-	 * @since 1.0.0
-	 *
-	 * @return void
-	 */
-	function _print_missing_build_admin_notice() {
-		?>
-		<div class="notice notice-error">
-			<p>
-				<strong><?php esc_html_e( 'Web Stories plugin could not be initialized.', 'web-stories' ); ?></strong>
-			</p>
-			<p>
-				<?php
-					echo wp_kses(
-						sprintf(
-						/* translators: %s: build commands. */
-							__( 'You appear to be running an incomplete version of the plugin. Please run %s to finish installation.', 'web-stories' ),
-							'<code>composer install &amp;&amp; npm install &amp;&amp; npm run build</code>'
-						),
-						[
-							'code' => [],
-						]
-					);
-				?>
-			</p>
-		</div>
-		<?php
-	}
+// Load Compatibility class the old fashioned way.
+if ( ! class_exists( '\Google\Web_Stories\Compatibility' ) ) {
+	require_once WEBSTORIES_PLUGIN_DIR_PATH . 'includes/Compatibility.php';
+}
+
+global $web_stories_compatibility;
+
+$web_stories_error = new WP_Error();
+$extensions        = [
+	'date'   => [
+		'classes' => [
+			'DateTimeImmutable',
+		],
+	],
+	'dom'    => [
+		'classes' => [
+			'DOMAttr',
+			'DOMComment',
+			'DOMDocument',
+			'DOMElement',
+			'DOMNode',
+			'DOMNodeList',
+			'DOMText',
+			'DOMXPath',
+		],
+	],
+	'json'   => [
+		'functions' => [
+			'json_decode',
+			'json_encode',
+		],
+	],
+	'libxml' => [
+		'functions' => [
+			'libxml_use_internal_errors',
+		],
+	],
+	'spl'    => [
+		'functions' => [
+			'spl_autoload_register',
+		],
+	],
+];
+
+$web_stories_compatibility = new Compatibility( $web_stories_error );
+$web_stories_compatibility->set_extensions( $extensions );
+$web_stories_compatibility->set_php_version( WEBSTORIES_MINIMUM_PHP_VERSION );
+$web_stories_compatibility->set_wp_version( WEBSTORIES_MINIMUM_WP_VERSION );
+$web_stories_compatibility->set_js_path( WEBSTORIES_PLUGIN_DIR_PATH . '/assets/js/edit-story.js' );
```

Comment:
`js_path` seems quite specific. What about `set_required_files` / `get_required_files` that accept an array of files?

Reply:
Yeah, I was thinking about that. If we have an array, should we check that these file exists? 

Should we also check for these files? 

https://github.com/google/web-stories-wp/blob/34d6e7f2e49c32b6ca9a8759d6e65952d899f430/web-stories.php#L63-L71

Reply:
Hmm the autoloader files would be covered by `check_php_built` I think? The plugin class won't exist without these files existing.

Reply:
It is likely that is the case, but something may go weird. I also think that scoper-autoloader is a little different from the autoloader. You may have one and not the other or something.

Reply:
The amp plugin does these checks. 

https://github.com/ampproject/amp-wp/blob/7b4272479c7c690138cf578e4b4f368edc15749d/amp.php#L155